api.py

```diff
@@ -157,6 +157,20 @@ def handle_webhook():
     else:
         return jsonify({'message': 'Invalid data format'}), 400
 
+def transform_gitlab_url(url):
+    # 去掉 http:// 或 https://
+    if url.startswith("http://"):
+        url = url[len("http://"):]
+    elif url.startswith("https://"):
+        url = url[len("https://"):]
+
+    if url.endswith('/'):
+        url = url[:-1]
+        
+    # 替换 . 和 / 为 _
+    transformed_url = url.replace('.', '_').replace('/', '_')
+    
+    return transformed_url
```

Comment:
quick suggestion to handle more edge cases with less lines of code:

```suggestion
def slugify_url(original: str) -> str:
    # Remove URL scheme (http, https, etc.) if present
    original = re.sub(r'^https?://', '', original)

    # Replace non-alphanumeric characters (except underscore) with underscores
    target = re.sub(r'[^a-zA-Z0-9]', '_', original)

    # Remove trailing underscore if present
    target = target.rstrip('_')

    return target
```